File: app/src/main/java/com/freestar/android/sample/recyclerview/Item.java
package com.freestar.android.sample.recyclerview;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Item implements Parcelable {

   @SerializedName("albumId")
   @Expose
   private Integer albumId;
   @SerializedName("id")
   @Expose
   private Integer id;
   @SerializedName("title")
   @Expose
   private String title;
   @SerializedName("url")
   @Expose
   private String url;
   @SerializedName("thumbnailUrl")
   @Expose
   private String thumbnailUrl;
   public final static Creator<Item> CREATOR = new Creator<Item>() {

      @SuppressWarnings({"unchecked"})
      public Item createFromParcel(Parcel in) {
         return new Item(in);
      }

      public Item[] newArray(int size) {
         return (new Item[size]);
      }

   };

   protected Item(Parcel in) {
      this.albumId = ((Integer) in.readValue((Integer.class.getClassLoader())));
      this.id = ((Integer) in.readValue((Integer.class.getClassLoader())));
      this.title = ((String) in.readValue((String.class.getClassLoader())));
      this.url = ((String) in.readValue((String.class.getClassLoader())));
      this.thumbnailUrl = ((String) in.readValue((String.class.getClassLoader())));
   }

   public Item() {
   }

   public Integer getAlbumId() {
      return albumId;
   }

   public void setAlbumId(Integer albumId) {
      this.albumId = albumId;
   }

   public Integer getId() {
      return id;
   }

   public void setId(Integer id) {
      this.id = id;
   }

   public String getTitle() {
      return title;
   }

   public void setTitle(String title) {
      this.title = title;
   }

   public String getUrl() {
      return url;
   }

   public void setUrl(String url) {
      this.url = url;
   }

   public String getThumbnailUrl() {
      return thumbnailUrl;
   }

   public void setThumbnailUrl(String thumbnailUrl) {
      this.thumbnailUrl = thumbnailUrl;
   }

   public void writeToParcel(Parcel dest, int flags) {
      dest.writeValue(albumId);
      dest.writeValue(id);
      dest.writeValue(title);
      dest.writeValue(url);
      dest.writeValue(thumbnailUrl);
   }

   public int describeContents() {
      return 0;
   }

}

File: Kotlin-Sample/app/src/main/java/com/freestar/sample/kotlin/Item.kt
package com.freestar.sample.kotlin

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import com.google.gson.annotations.Expose
import android.os.Parcel

class Item : Parcelable {
    @SerializedName("albumId")
    @Expose
    var albumId: Int? = null

    @SerializedName("id")
    @Expose
    var id: Int? = null

    @SerializedName("title")
    @Expose
    var title: String? = null

    @SerializedName("url")
    @Expose
    var url: String? = null

    @SerializedName("thumbnailUrl")
    @Expose
    var thumbnailUrl: String? = null

    protected constructor(`in`: Parcel) {
        albumId = `in`.readValue(Int::class.java.classLoader) as Int?
        id = `in`.readValue(Int::class.java.classLoader) as Int?
        title = `in`.readValue(String::class.java.classLoader) as String?
        url = `in`.readValue(String::class.java.classLoader) as String?
        thumbnailUrl = `in`.readValue(String::class.java.classLoader) as String?
    }

    constructor() {}

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(albumId)
        dest.writeValue(id)
        dest.writeValue(title)
        dest.writeValue(url)
        dest.writeValue(thumbnailUrl)
    }

    override fun describeContents(): Int {
        return 0
    }

    /*companion object {
        val CREATOR: Parcelable.Creator<Item> = object : Parcelable.Creator<Item?> {
            override fun createFromParcel(`in`: Parcel): Item? {
                return Item(`in`)
            }

            override fun newArray(size: Int): Array<Item?> {
                return arrayOfNulls(size)
            }
        }
    }*/

    companion object CREATOR : Parcelable.Creator<Item> {
        override fun createFromParcel(parcel: Parcel): Item {
            return Item(parcel)
        }

        override fun newArray(size: Int): Array<Item?> {
            return arrayOfNulls(size)
        }
    }
}